arkouda/io.py

```diff
@@ -444,7 +466,9 @@ def _build_objects(
     # 2. We have a single pdarray
     # 3. We have a single strings object
     if len(items) > 1:  # DataSets condition
-        return {item["dataset_name"]: _parse_obj(item) for item in items}
+        ds_dict = _dict_recombine_segarrays({item["dataset_name"]: _parse_obj(item) for item in items})
+        # if dict only has 1 element when it had >1 before, the element must be a segarray
+        return next(iter(ds_dict.values())) if len(ds_dict.keys()) == 1 else ds_dict
```

Comment:
Noting when SegArray IO updates are pushed, this can be reverted to 

```python
return {item["dataset_name"]: _parse_obj(item) for item in items}
```